main.go

```diff
@@ -652,7 +567,7 @@ func main() {
 		}
 		cacheModel.FilePathMap = filePathsMap
 
-		recacheRequired, err = cacheModel.CompareFilePathMaps(filePathsMap)
+		recacheRequired = CompareFilePathMaps(cacheModel.PreviousFilePathMap, filePathsMap, cacheModel.DebugMode)
```

Comment:
The new `CompareFilePathMaps()` method will not return error, but you still check the `err != nil` 
In this case it's the previous error which was already checked at:
https://github.com/bitrise-steplib/steps-cache-push/blob/2a5944daa914ebf9fc6f83a7549d9b12b99a0e63/main.go#L564

Reply:
another good catch